server/item/enchantment/enchantment.go

```diff
@@ -0,0 +1,6 @@
+package enchantment
+
+// Curse represents an enchantment that may be a curse enchantment.
+type Curse interface {
```

Comment:
You can move this interface into the session package under `handler_grindstone.go`. Please also rename to `curseEnchantment` for consistency with `treasureEnchantment`.